specs/commandline/sign.md

```diff
@@ -0,0 +1,83 @@
+# notation sign
+
+## Description
+
+Use `notation sign` to sign artifacts.
+
+Signs a container artifact that is stored in a registry. Upon successful signing, the generated signature is pushed to the registry and the digest of the container image is returned.
+
+## Outline
+
+```text
+Sign artifacts
+
+Usage:
+  notation sign <reference> [flags]
+
+Flags:
+      --cert-file string        Location of file containing a complete certificate chain for the signing key. Use this flag with '--key-file'.
+  -e, --expiry duration         Optional expiry that provides a "best by use" time for the artifact. The duration is specified in minutes(m), hours(h) or days(d). For example: 30d, 12h, 30m, 1d3h20m
+  -h, --help                    Help for sign
+  -k, --key string              Signing key name, for a key previously added to notation's key list.
+      --key-file string         Location of file containing signing key file. Use this flag with '--cert-file'.
+  -p, --password string         Password or identity token for registry operations (default to $NOTATION_PASSWORD if not specified)
+  --plugin-config strings       List of {key}={value} pairs that are passed as is to a plugin, if the key (--key) is associated with a signing plugin, refer plugin documentation to set appropriate values
+  -u, --username string         Username for registry operations (default to $NOTATION_USERNAME if not specified)
+
+Global Flags:
+      --plain-http   Registry access via plain HTTP
+```
+
+## Usage
+
+### Sign a container image
+
+```shell
+# Add a key which uses a local private key and certificate, and make it a default signing key
+notation key add --name <key_name> <key_path> <cert_path> --default
```

Comment:
It is better to put all flags before arguments although they can be put after the arguments.